### DISCUSSION TITLE
Question about aihwkit install

### DISCUSSION THREAD

**Discussion by YuqiL1** [IBM/aihwkit#582] [Installation Q&A]
## Description

I have met some bug when i am installing aihwkit on my computer, and i would like to use the cpu. Thus, I started by following the installation instructions. I have typically installed requirements by:
$ pip install -r requirements.txt
$ pip install -r requirements-dev.txt
$ pip install -r requirements-examples.txt
and then i use' pip  install awhikit', I get the output:

[pip install.txt](https://github.com/IBM/aihwkit/files/13749449/pip.install.txt)

And then I use 'python setup.py build_ext --inplace -DUSE_CUDA=OFF'. I get the output:

--------------------------------------------------------------------------------
-- Trying 'Ninja (Visual Studio 17 2022 x64 v143)' generator
--------------------------------
---------------------------
----------------------
-----------------
------------
-------
--
Not searching for unused variables given on the command line.
CMake Deprecation Warning at CMakeLists.txt:1 (cmake_minimum_required):
  Compatibility with CMake < 3.5 will be removed from a future version of
  CMake.

  Update the VERSION argument <min> value or use a ...<max> suffix to tell
  CMake that the project does not need compatibility with older versions.

-- The C compiler identification is MSVC 19.38.33133.0
-- Detecting C compiler ABI info
-- Detecting C compiler ABI info - done
-- Check for working C compiler: C:/Program Files (x86)/Microsoft Visual Studio/2022/BuildTools/VC/Tools/MSVC/14.38.33130/bin/Hostx86/x64/cl.exe - skipped
-- Detecting C compile features
-- Detecting C compile features - done
-- The CXX compiler identification is MSVC 19.38.33133.0
-- Detecting CXX compiler ABI info
-- Detecting CXX compiler ABI info - done
-- Check for working CXX compiler: C:/Program Files (x86)/Microsoft Visual Studio/2022/BuildTools/VC/Tools/MSVC/14.38.33130/bin/Hostx86/x64/cl.exe - skipped
-- Detecting CXX compile features
-- Detecting CXX compile features - done
-- Configuring done (1.5s)
-- Generating done (0.0s)
-- Build files have been written to: E:/aihwkit/_cmake_test_compile/build
--
-------
------------
-----------------
----------------------
---------------------------
--------------------------------
-- Trying 'Ninja (Visual Studio 17 2022 x64 v143)' generator - success
--------------------------------------------------------------------------------

Configuring Project
  Working directory:
    E:\aihwkit\_skbuild\win-amd64-3.9\cmake-build
  Command:
    'D:\BaiduNetdiskDownload\envs\pcm\lib\site-packages\cmake\data\bin/cmake.exe' 'E:\aihwkit' -G Ninja -D_SKBUILD_FORCE_MSVC=1930 --no-warn-unused-cli '-DCMAKE_INSTALL_PREFIX:PATH=E:\aihwkit\_skbuild\win-amd64-3.9\cmake-install' -DPYTHON_VERSION_STRING:STRING=3.9.0 -DSKBUILD:INTERNAL=TRUE '-DCMAKE_MODULE_PATH:PATH=D:\BaiduNetdiskDownload\envs\pcm\lib\site-packages\skbuild\resources\cmake' '-DPYTHON_EXECUTABLE:PATH=D:\BaiduNetdiskDownload\envs\pcm\python.exe' '-DPYTHON_INCLUDE_DIR:PATH=D:\BaiduNetdiskDownload\envs\pcm\Include' '-DPYTHON_LIBRARY:PATH=D:\BaiduNetdiskDownload\envs\pcm\libs\python39.lib' '-DPython_EXECUTABLE:PATH=D:\BaiduNetdiskDownload\envs\pcm\python.exe' '-DPython_ROOT_DIR:PATH=D:\BaiduNetdiskDownload\envs\pcm' -DPython_FIND_REGISTRY:STRING=NEVER '-DPython_INCLUDE_DIR:PATH=D:\BaiduNetdiskDownload\envs\pcm\Include' '-DPython_LIBRARY:PATH=D:\BaiduNetdiskDownload\envs\pcm\libs\python39.lib' '-DPython_NumPy_INCLUDE_DIRS:PATH=D:\BaiduNetdiskDownload\envs\pcm\lib\site-packages\numpy\core\include' '-DPython3_EXECUTABLE:PATH=D:\BaiduNetdiskDownload\envs\pcm\python.exe' '-DPython3_ROOT_DIR:PATH=D:\BaiduNetdiskDownload\envs\pcm' -DPython3_FIND_REGISTRY:STRING=NEVER '-DPython3_INCLUDE_DIR:PATH=D:\BaiduNetdiskDownload\envs\pcm\Include' '-DPython3_LIBRARY:PATH=D:\BaiduNetdiskDownload\envs\pcm\libs\python39.lib' '-DPython3_NumPy_INCLUDE_DIRS:PATH=D:\BaiduNetdiskDownload\envs\pcm\lib\site-packages\numpy\core\include' -DUSE_CUDA=OFF -DCMAKE_BUILD_TYPE:STRING=Release

Not searching for unused variables given on the command line.
-- Invoking cmake through scikit-build
-- The BLAS backend of choice:OpenBLAS
-- Could not find OpenBLAS include. Turning OpenBLAS_FOUND off
-- Could not find OpenBLAS lib. Turning OpenBLAS_FOUND off
CMake Error at cmake/Modules/FindOpenBLAS.cmake:89 (MESSAGE):
  Could not find OpenBLAS
Call Stack (most recent call first):
  cmake/dependencies.cmake:56 (find_package)
  CMakeLists.txt:48 (include)

-- Configuring incomplete, errors occurred!
Traceback (most recent call last):
  File "D:\BaiduNetdiskDownload\envs\pcm\lib\site-packages\skbuild\setuptools_wrap.py", line 666, in setup
    env = cmkr.configure(
  File "D:\BaiduNetdiskDownload\envs\pcm\lib\site-packages\skbuild\cmaker.py", line 357, in configure
    raise SKBuildError(msg)

An error occurred while configuring with CMake.
  Command:
    'D:\BaiduNetdiskDownload\envs\pcm\lib\site-packages\cmake\data\bin/cmake.exe' 'E:\aihwkit' -G Ninja -D_SKBUILD_FORCE_MSVC=1930 --no-warn-unused-cli '-DCMAKE_INSTALL_PREFIX:PATH=E:\aihwkit\_skbuild\win-amd64-3.9\cmake-install' -DPYTHON_VERSION_STRING:STRING=3.9.0 -DSKBUILD:INTERNAL=TRUE '-DCMAKE_MODULE_PATH:PATH=D:\BaiduNetdiskDownload\envs\pcm\lib\site-packages\skbuild\resources\cmake' '-DPYTHON_EXECUTABLE:PATH=D:\BaiduNetdiskDownload\envs\pcm\python.exe' '-DPYTHON_INCLUDE_DIR:PATH=D:\BaiduNetdiskDownload\envs\pcm\Include' '-DPYTHON_LIBRARY:PATH=D:\BaiduNetdiskDownload\envs\pcm\libs\python39.lib' '-DPython_EXECUTABLE:PATH=D:\BaiduNetdiskDownload\envs\pcm\python.exe' '-DPython_ROOT_DIR:PATH=D:\BaiduNetdiskDownload\envs\pcm' -DPython_FIND_REGISTRY:STRING=NEVER '-DPython_INCLUDE_DIR:PATH=D:\BaiduNetdiskDownload\envs\pcm\Include' '-DPython_LIBRARY:PATH=D:\BaiduNetdiskDownload\envs\pcm\libs\python39.lib' '-DPython_NumPy_INCLUDE_DIRS:PATH=D:\BaiduNetdiskDownload\envs\pcm\lib\site-packages\numpy\core\include' '-DPython3_EXECUTABLE:PATH=D:\BaiduNetdiskDownload\envs\pcm\python.exe' '-DPython3_ROOT_DIR:PATH=D:\BaiduNetdiskDownload\envs\pcm' -DPython3_FIND_REGISTRY:STRING=NEVER '-DPython3_INCLUDE_DIR:PATH=D:\BaiduNetdiskDownload\envs\pcm\Include' '-DPython3_LIBRARY:PATH=D:\BaiduNetdiskDownload\envs\pcm\libs\python39.lib' '-DPython3_NumPy_INCLUDE_DIRS:PATH=D:\BaiduNetdiskDownload\envs\pcm\lib\site-packages\numpy\core\include' -DUSE_CUDA=OFF -DCMAKE_BUILD_TYPE:STRING=Release
  Source directory:
    E:\aihwkit
  Working directory:
    E:\aihwkit\_skbuild\win-amd64-3.9\cmake-build
Please see CMake's output for more information.

 I do not konw how to address them , please help me!!!!

## How to reproduce

None

## Expected behavior

install aihwkit

## Other information

None

* Pytorch version: requirements
* Package version:requirements
* OS:
* Python version:3.9
* Conda version (or N/A):

**Comment by maljoras** [comment]
Hi @YuqiL1,
You need to install either openblas or mkl to be able to compile it. Please take a look at the [installation instructions](https://aihwkit.readthedocs.io/en/latest/advanced_install.html#windows-with-openblas-experimental).

**Comment by jubueche** [comment]
@YuqiL1 did you manage to install it?

**Comment by Raghucharan16** [comment]
when i'm installing aihwkit via pip install command i'm getting _lzma module not found. I was trying to setup a nn using aihwkit on mnist data. so how can properly install it or which python version perfectly supports it. I'm getting torch and torchvision dependency errors since aihwkit needs torch 2.0.1 but torchvision needs torch 2.3 and if all were installed i'm getting _lzma module not found. I also installed lzma in my ubuntu by sudp apt install aihwkit.

**Reply by maljoras-sony** [reply]
There is no torch version requirements really. Maybe try to compile aihwkit directly from the repository. It should work just fine with torch 2.3
